File: seleniumCSharp/Reqnroll/Practice/HoangTramHuynh/Page/HomePage.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;
using HoangTramHuynh.Core;
using OpenQA.Selenium;
namespace HoangTramHuynh.Page
{
    public class HomePage
    {
        private WebObject _dropdownproject = new WebObject(By.XPath("//a[contains(.,'Projects')]"), "Dropdown Project");
        private WebObject _createproject = new WebObject(By.CssSelector("a[data-target='#modalCreateProject']"), "Create Project Option");
        private WebObject _searchproject = new WebObject(By.XPath("//a[contains(.,'Search Project')]"), "Search Project Option");
        private WebObject _avatar = new WebObject(By.CssSelector("img#ava"), "Avatar account login");
        private WebObject accountlogininformation(string username)
        {
            return new WebObject(
                By.XPath($"//img[@id='ava']/ancestor::a[contains(normalize-space(.),'{username}')]"),
                "Account Login Information"
            );
        }
        public bool IsHomePageDisplayed()
        {
            return _avatar.IsElementDisplayed();
        }

        public bool IsLoginWithCorrectAccount(string username)
        {
            return accountlogininformation(username).IsElementDisplayed();
        }

        public bool IsProjectDropdownDisplayed()
        {
            return _dropdownproject.IsElementDisplayed();
        }

        public bool IsCreateProjectOptionDisplayed()
        {
            return _createproject.IsElementDisplayed();
        }

        public bool IsSearchProjectOptionDisplayed()
        {
            return _searchproject.IsElementDisplayed();
        }

        public void ClickProjectDropdown()
        {
            _dropdownproject.ClickOnElement();
        }

        public void ClickCreateProject()
        {
            _createproject.ClickOnElement();
        }

        public void ClickSearchProject()
        {
            _searchproject.ClickOnElement();
        }


        public void NavigateToCreateProject()
        {
            ClickProjectDropdown();

            ClickCreateProject();
        }

        public void NavigateToSearchProject()
        {
            ClickProjectDropdown();

            ClickSearchProject();
        }

    }
}

File: playwright/practice/HoangTramHuynh-PW/page-object/home-page.ts
import { Locator, Page } from '@playwright/test';

export class HomePage {
  private readonly page: Page;
  private readonly avatar: Locator;

  constructor(page: Page) {
    this.page = page;
    this.avatar = page.locator('img#ava');
  }

  private accountLoginInformation(username: string): Locator {
    return this.page.locator(
      `//img[@id='ava']/ancestor::a[contains(normalize-space(.),'${username}')]`
    );
  }

    async verifyHomePageIsDisplayed(): Promise<void> {
    await this.avatar.waitFor({ state: 'visible', timeout: 15000 });
    }

    async verifyLoginWithCorrectAccount(username: string): Promise<void> {
    await this.accountLoginInformation(username).waitFor({
        state: 'visible',
        timeout: 15000,
    });
    }

}